pkg/controller/core/workload_controller.go

```diff
@@ -95,6 +100,15 @@ func WithWorkloadUpdateWatchers(value ...WorkloadUpdateWatcher) Option {
 	}
 }
 
+// WithWorkloadRetention allows to specify retention for workload resources
+func WithWorkloadRetention(value *workloadRetentionConfig) Option {
+	return func(o *options) {
+		if value != nil && value.afterFinished != nil {
+			o.workloadRetentionConfig.afterFinished = value.afterFinished
+		}
```

Comment:
Actually, can we just say `o.workloadRetentionConfig= value`? It would simplify the code. Currently there is a mismatch: function name is generic, but we only copy the `afterFinished`.